### PR TITLE
Allow to rename archive policy rules

### LINKED ISSUE #374: Allow to rename archive policy rules
Currently, the archive policy rules cannot be renamed. It would be cool to allow them to be renamed.

### PR DESCRIPTION
**Pull request by 0livd**
It is now possible to change the name of an archive policy rule.

Closes #374

### CODE REVIEW

File: doc/source/rest.j2 [hunk 1]
```diff
@@ -300,6 +300,10 @@ It is possible to delete an archive policy rule:
 
 {{ scenarios['delete-archive-policy-rule']['doc'] }}
 
+It is possible to remane an archive policy rule:
```
Comment (jd) [inline doc/source/rest.j2]: rename

Comment (0livd) [reply doc/source/rest.j2]: Corrected :)

File: gnocchi/tests/functional/gabbits/archive-rule.yaml [hunk 2]
```diff
@@ -158,6 +158,32 @@ tests:
...
+
+    - name: rename archive policy rule with existing name
+      PATCH: /v1/archive_policy_rule/test_rule2
+      request_headers:
+        # User admin
+        authorization: "basic YWRtaW46"
+      data:
+          name: test_rule3_renamed
+      status: 400
+
```
Comment (chungg) [inline gnocchi/tests/functional/gabbits/archive-rule.yaml]: could you maybe add a check to verify test_rule3 (original) doesn't exist after rename

Comment (0livd) [reply gnocchi/tests/functional/gabbits/archive-rule.yaml]: Done, added a test with missing name parameter too

**Comment by jd** [comment]
It'd be great to squash all of those in a single commit, but ok.

File: gnocchi/tests/functional/gabbits/archive-rule.yaml [hunk 3]
```diff
@@ -158,6 +158,45 @@ tests:
...
```
Comment (jd) [inline gnocchi/tests/functional/gabbits/archive-rule.yaml]: would be great to check for the reason of that status 400.

**Comment by 0livd** [comment]
Ah sorry I just rebased the whole thing but it dismissed your reviews...I have to check out how to use pastamaker...

File: gnocchi/tests/functional/gabbits/archive-rule.yaml (hunk 3 above)
Comment (0livd) [reply gnocchi/tests/functional/gabbits/archive-rule.yaml]: I added a check for the response string

**Comment by jd** [comment]
Yeah it's normal that reviews disappear don't worry. :)

**Pull request merged**